File: src/main/java/technology/touchmars/feign/wechat/client/config/WechatClientException.java
/**
 * 
 */
package technology.touchmars.feign.wechat.client.config;

import feign.codec.DecodeException;

/**
 * @author jhcao
 *
 */
@SuppressWarnings("serial")
public class WechatClientException extends DecodeException {

	private int errorCode;
	
	public WechatClientException(SimpleResponse modelApiResponse) {
		super(modelApiResponse.getErrmsg());
		this.errorCode = modelApiResponse.getErrcode();
	}
	
	public WechatClientException(int errCode, String errMsg){
		super(errMsg);
		this.errorCode = errCode;
	}

	public int getErrorCode() {
		return errorCode;
	}

	@Override
	public String toString(){
		return String.format("WechatClientException::\n errorCode = %s, errorMsg = %s", errorCode, getMessage());
	}
}

File: src/main/kotlin/technology/touchmars/feign/wechat/client/config/WechatClientException.kt
/**
 *
 */
package technology.touchmars.feign.wechat.client.config

import feign.codec.DecodeException

/**
 * @author jhcao
 */
@SuppressWarnings("serial")
class WechatClientException : DecodeException {

    var errorCode: Int? = 0
        private set

    constructor(modelApiResponse: SimpleResponse) : super(modelApiResponse.errmsg) {
        this.errorCode = modelApiResponse.errcode
    }

    constructor(errCode: Int, errMsg: String) : super(errMsg) {
        this.errorCode = errCode
    }

    @Override
    override fun toString(): String {
        return String.format("WechatClientException::\n errorCode = %s, errorMsg = %s", errorCode, message)
    }
}
